### PR TITLE
Allow key-value Environment variable passthrouugh.

### PR DESCRIPTION
**Pull request by jdmaguire**
### Motivation

`.spec.secretsMap` exists to pass some configuration to the sinks/sources/functions. Using this for plain, simple configuration is ambiguous/unintuitive and is more complicated since it requires multiple K8s resources (i.e. the Function CR and a k8s Secret).

I'm not deeply familiar with this project. Feel free to suggest alternatives or dismiss it as you see fit.

### Modifications

Introduces `.spec.env` that gets combined with `.spec.secretsMap` to form the environment variables for the statefulset/pods deployed by this.

### Verifying this change

I verified this change by manually creating the CRs, `kubectl exec`'ing in and verifying that things work w/o `.spec.env` as before and with `.spec.env`. I should add more tests but I wasn't familiar with this codebase so I wanted to write the code and get the PR up first.

### Documentation

Check the box below.

Need to update docs? 

- [] `doc-required` 

- [ ] `no-need-doc` 

- [x] `doc`

### CODE REVIEW

**Review by tpiperatgod — APPROVED** [review]
lgtm

**Review by jiangpengcheng — APPROVED** [review]

**Review by freeznet — APPROVED** [review]

File: api/v1alpha1/function_types.go [hunk 1]
```diff
@@ -51,6 +51,7 @@ type FunctionSpec struct {
 	FuncConfig   *Config                     `json:"funcConfig,omitempty"`
 	Resources    corev1.ResourceRequirements `json:"resources,omitempty"`
 	SecretsMap   map[string]SecretRef        `json:"secretsMap,omitempty"`
+	Env          map[string]string           `json:"env,omitempty"`
```
Comment (freeznet) [inline api/v1alpha1/function_types.go:54]: Is it okay if move this `Env` part to `PodPolicy` struct? and it is suggested to use `[]corev1.EnvVar` so it is aligned with Kubernetes APIs.

**Review by freeznet — COMMENTED** [review]
@jdmaguire thanks for the PR, please check my comments, thanks.

File: api/v1alpha1/function_types.go (hunk 1 above)
Comment (jdmaguire) [reply api/v1alpha1/function_types.go:54]: Thanks for the suggestion. I was actually thinking of doing that but was unsure how y'all would feel about it.

I'll switch to that.

**Pull request merged**